python/src/lib.rs

```diff
@@ -15,18 +15,189 @@
+    */
     Ok(())
 }
+
+// Ballista Builder will take a HasMap/Dict Cionfg
+#[pyclass(name = "BallistaBuilder", module = "ballista", subclass)]
+pub struct PyBallistaBuilder {
+    conf: HashMap<String, String>,
+}
+
+#[pymethods]
+impl PyBallistaBuilder {
+    #[new]
+    pub fn new() -> Self {
+        Self { conf: HashMap::new() }
+    }
+
+    pub fn set(
+        mut slf: PyRefMut<'_, Self>,
+        k: &str,
+        v: &str,
+        py: Python,
+    ) -> PyResult<PyObject> {
+        slf.conf.insert(k.into(), v.into());
+
+        Ok(slf.into_py(py))
+    }
+
+    pub fn show_config(&self) {
+        println!("Ballista Config:");
+        for ele in self.conf.iter() {
+            println!("\t{}: {}", ele.0, ele.1)
+        }
+    }
+
+    pub fn build(slf: PyRef<'_, Self>) -> PyBallista {
+        PyBallista {
+            conf: PyBallistaBuilder {
+                conf: slf.conf.clone(),
+            },
+        }
+    }
+}
+
+#[pyclass(name = "Ballista", module = "ballista", subclass)]
+pub struct PyBallista {
+    pub conf: PyBallistaBuilder,
+}
+
+#[pymethods]
+impl PyBallista {
+    #[new]
+    pub fn new() -> Self {
+        Self {
+            conf: PyBallistaBuilder::new(),
+        }
+    }
+
+    pub fn show_config(&self) {
+        println!("Ballista Config:");
+        for ele in self.conf.conf.clone() {
+            println!("{:4}: {}", ele.0, ele.1)
+        }
+    }
+
+    /// Construct the standalone instance from the SessionContext
+    #[pyo3(signature = (concurrent_tasks = 4))]
+    pub fn standalone(
+        &self,
+        concurrent_tasks: usize,
+        py: Python,
+    ) -> PyResult<DataFusionPythonSessionContext> {
+        // Build the config
+        let config = &BallistaConfig::with_settings(self.conf.conf.clone()).unwrap();
+        // Define the SessionContext
+        let session_context = BallistaContext::standalone(&config, concurrent_tasks);
+        // SessionContext is an async function
+        let ctx = wait_for_future(py, session_context)
+            .map_err(to_pyerr)?
+            .context()
+            .clone();
+
+        // Convert the SessionContext into a Python SessionContext
+        Ok(ctx.into())
+    }
+
+    /// Construct the remote instance from the SessionContext
+    pub fn remote(
+        &self,
+        host: &str,
+        port: u16,
+        py: Python,
+    ) -> PyResult<DataFusionPythonSessionContext> {
+        // Build the config
+        let config = &BallistaConfig::with_settings(self.conf.conf.clone()).unwrap();
+        // Create the BallistaContext
+        let session_context = BallistaContext::remote(host, port, config);
+        let ctx = wait_for_future(py, session_context)
+            .map_err(to_pyerr)?
+            .context()
+            .clone();
+
+        // Convert the SessionContext into a Python SessionContext
+        Ok(ctx.into())
+    }
+}
+
+/*
+Plan to implement Session Config and State in a future issue
```

Comment:
I don't see reason for `PySessionConfig` why do you think it is important?
Can we get it from datafusion-python

Reply:
Also, please remove comment if it is dead code

Reply:
removed dead code